lib/new_relic/agent/instrumentation/grape.rb

```diff
@@ -19,9 +19,14 @@
 
   depends_on do
     begin
-      if defined?(Bundler) && Bundler.rubygems.all_specs.map(&:name).include?('newrelic-grape')
-        NewRelic::Agent.logger.info('Not installing New Relic supported Grape instrumentation because the third party newrelic-grape gem is present')
-        false
+      if defined?(Bundler) &&
+          if Bundler::VERSION >= '2' && Bundler.rubygems.installed_specs.map(&:name).include?('newrelic-grape')
+            NewRelic::Agent.logger.info('Not installing New Relic supported Grape instrumentation because the third party newrelic-grape gem is present')
+            false
+          elsif Bundler.rubygems.all_specs.map(&:name).include?('newrelic-grape')
+            NewRelic::Agent.logger.info('Not installing New Relic supported Grape instrumentation because the third party newrelic-grape gem is present')
+            false
+          end
```

Comment:
Two problems here:

1. You don't have an `else` for Bundler 2 being present and `newrelic-grape` being absent. The `else` should yield `true`.
2. Both the `if` and `elsif` do the exact same 2 things (logger statement and yield `false`), so you can just `||` them together and DRY up the code.

Here's a pass at addressing both issues:

```suggestion
          grape_present = if (Bundler::VERSION >= '2' && Bundler.rubygems.installed_specs.map(&:name).include?('newrelic-grape')) ||
                            Bundler.rubygems.all_specs.map(&:name).include?('newrelic-grape')
                            true
                          else
                            false
                          end
          if grape_present
            NewRelic::Agent.logger.info('Not installing New Relic supported Grape instrumentation because the third party newrelic-grape gem is present')
            false
          else
            true
          end
```

Reply:
Much much cleaner. I don't think I was looking at the code holistically. Great edit.